### PR TITLE
Add more tests to increase code coverage

### PR DESCRIPTION
**Pull request by PoLaKoSz**
Probably this is a huge commit but it does not contain any breaking changes (only adds more test cases to the current Unit test project).

Additionally I added two different type of tests called `Integration` and `Regression`.
The **Integration** tests just loads a previously saved API response and tries to parse it, while the **Regression** tests calls directly the live API.

I want to mention that this commit will introduce 28 failing tests (but increase the code coverage from **~41%** to **~68%**). I already implemented the fixes for them but i thought i will wait for some feedback first.

Ref.: #26

### CODE REVIEW

File: tests/Regression/Endpoint/UserEndpointTests.cs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    [TestFixture]
+    class UserEndpointTests
+    {
+        private static IUserEndpoint _user;
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            var deezer = DeezerSession.CreateNew();
+            string token = "";
```
Comment (PoLaKoSz) [inline tests/Regression/Endpoint/UserEndpointTests.cs:24]: Currently this is where a valid API token should be added before running the `Regression` tests.

**Comment by projectgoav** [comment]
Thanks to submitting this. I'll review it over the next few days (as I'm away) and get back to you! :)

**Comment by PoLaKoSz** [comment]
I added an another commit that i forgot yesterday which adds more live API call tests (Add and Remove from Favourites - Artist, Track, Album, Playlist and Radio).
I wasn't able to test what property change when i rate an album, so i didn't added a test for it and also i didn't added test to post a comment because it maybe trigger a notification and i don't want to spam other users with notifications.

With this the failing tests count increase to 46 but with the fixes it will increase the code coverage too with additional **~5%**.

I wanted to ask should i commit the fixes here too? It contains two breaking change:
- `Album`'s `GenreId` type changes from `uint` to `int`
- `BrowseEndpoint`'s `GetUserById` return type changes from `Task<IUser>` to `Task<IUserProfile>` . I changed the type because the return json is more similar to this type compared to the `IUser`. 
   I want to mention that i added a new property called `Country` to the `IUserProfile` because a `GET` request to the https://api.deezer.com/user/5 URL will return a `country` property too but maybe this is not the best option because i saw for example the https://api.deezer.com/album/302127/fans will not contains the `country` property. What do You think about this?

**Comment by PoLaKoSz** [comment]
Did You have a chance to take a look at it?

File: tests/Integration/StaticResources/user.me.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿{
+  "id": "1518367622",
+  "name": "PoLáKoSz",
+  "lastname": "",
+  "firstname": "",
+  "email": "polakosz@freemail.hu",
```
Comment (projectgoav) [inline tests/Integration/StaticResources/user.me.json:6]: I hope this isn't your real email!

**Review by projectgoav — CHANGES_REQUESTED** [review]
Thanks for doing this work 👍  

Looks good and sorry for not getting to this sooner.  Assuming it's not your (or a valid) email in some of the JSON I'll be happy to merge this.

A couple of other points (mostly to remind me of things to do later):
- I'd prefer if there was a consistent coding style throughout the project. I'm not overly happy with the style used originally and have been slowly changing bits as I touch the code. I'll come up with a style and note it somewhere for anyone wishing to contribute in the future. 

- I *might* change the folder structure a little from how you have laid it out but again I'm not entirely sure how.

File: tests/Integration/StaticResources/user.me.json (hunk 2 above)
Comment (PoLaKoSz) [reply tests/Integration/StaticResources/user.me.json:6]: This is my valid email, but it's publicly available (including on my GitHub profile too) so i don't think this is a security issue. :)

**Comment by PoLaKoSz** [comment]
> * I'd prefer if there was a consistent coding style throughout the project. I'm not overly happy with the style used originally and have been slowly changing bits as I touch the code. I'll come up with a style and note it somewhere for anyone wishing to contribute in the future.

Usually just temporary but i often use a Microsoft tool called [StyleCop](https://github.com/DotNetAnalyzers/StyleCopAnalyzers) to address this issue. It has a great [documentation](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/DOCUMENTATION.md) which can help to modify the default rules (for ex.:
- brackets into new line
- fields first, constructor(s) than properties, methods
- etc.).

I wanted to ask should i commit the fixes here too? It contains two breaking change:
- `Album`'s `GenreId` type changes from `uint` to `int`
- `BrowseEndpoint`'s `GetUserById` return type changes from `Task<IUser>` to `Task<IUserProfile>` . I changed the type because the return json is more similar to this type compared to the `IUser`. 
   I want to mention that i added a new property called `Country` to the `IUserProfile` because a `GET` request to the https://api.deezer.com/user/5 URL will return a `country` property too but maybe this is not the best option because i saw for example the https://api.deezer.com/album/302127/fans will not contains the `country` property. What do You think about this?

File: tests/Integration/StaticResources/user.me.json (hunk 2 above)
Comment (projectgoav) [reply tests/Integration/StaticResources/user.me.json:6]: As long as you're happy for your email to go in, I'll get this merged.

Comment (PoLaKoSz) [reply tests/Integration/StaticResources/user.me.json:6]: It's not a problem for me :)

**Pull request merged**